lightrag/lightrag.py

```
@@ -305,33 +307,115 @@ async def _insert_done(self):
+                    for dp in all_relationships_data
+                }
+                await self.relationships_vdb.upsert(data_for_vdb)
+        finally:
+            if update_storage:
+                await self._insert_done()
+
     def query(self, query: str, param: QueryParam = QueryParam()):
         loop = always_get_an_event_loop()
         return loop.run_until_complete(self.aquery(query, param))
 
     async def aquery(self, query: str, param: QueryParam = QueryParam()):
```

Ruff (0.8.0)

Do not perform function call `QueryParam` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)